### ISSUE TITLE
whitelist does not working

### ISSUE THREAD

**Issue by ookangzheng** [pi-hole/pi-hole#675]
##

**Comment by PromoFaux** [comment]
Please elaborate.

**Comment by kochse** [comment]
I have the same issue. Cannot add item to the whitelist. Updated to the last version.

**Comment by PromoFaux** [comment]
So, is this on the web interface, or using `pihole -w` on the command line? 

Please be more descriptive!

**Comment by kochse** [comment]
Manipulating the lists via command line works. Add and delete via the web interface has no effect. I'll try to add more informations.

**Comment by PromoFaux** [comment]
OK, there have been some known issues on the web interface whitelists in the past. Can you try accessing the web interface with the address http://pi.hole/admin and trying again.

If it still fails, and you're using chrome, press `F12` to open the developer menu and then take a look at the network tab to see if there are any errors shown.

Possibly a duplicate of pi-hole/AdminLTE/issues/126

**Comment by PromoFaux** [comment]
@ookangzheng Is the issue you're facing with the Admin page, or on the command line?

**Comment by ookangzheng** [comment]
command line
i tried to add in admin page in whitelist or blacklist ~~ not working
My server
Ubuntu 16.04, Nginx, PHP-7

**Comment by Mcat12** [comment]
Please run `pihole -d`

**Comment by ookangzheng** [comment]
Last time I also tried in Centos 6 with nginx and php

**Comment by ookangzheng** [comment]
when I restart my server, the pihole service does not auto run automatically

**Comment by Mcat12** [comment]
Without that debug log, it will be very hard to help you, so please run `pihole -d`.

**Comment by ookangzheng** [comment]
http://termbin.com/52ig

**Comment by PromoFaux** [comment]
@ookangzheng That debug log says you're on Ubuntu 16.04, [but the other issue you just opened](https://github.com/pi-hole/pi-hole/issues/684) is for a centOS6 install. Have you changed distros?

as for this one, I'm noticing in the debug log that the WebUI version check is coming back blank.

Is there anything in the directory `/var/www/html/admin`? If so, could you run `git describe --tags --abbrev=0` in that directory and tell us the output, please?

Thanks

**Comment by gerard33** [comment]
Having the same on Ubuntu 14.04. Add and remove in black/whitelist from webinterface is not working. The interface gives a success message, but nothing is changed (also not in the files in `/etc/pihole`).
It was working with webinterface v1.3 (after updating to PHP7).
`git describe --tags --abbrev=0` in `/var/www/html/admin` gives: v1.4

**Comment by nomoreads** [comment]
Same issue on my end. I see a lot of different "whitelisting" issue threads around here.
I can add a domain to the whitelist from the webinterface but Pihole just ignores it, still blocking!

**Comment by PromoFaux** [comment]
Quoting @ookangzheng from closed duplicate issue #696 
>I use on Centos6
>I add pihole -w -f tieba.baidu.com , but it not working
>Error message: tieba.baidu.com refused to connect. (chrome)


OK, @ookangzheng... a couple of things i've noticed in your debug log...

You've added the following lists:
```
https://github.com/AdguardTeam/AdguardDNS/raw/master/Filters/filter.txt
https://raw.githubusercontent.com/cjx82630/cjxlist/master/cjx-annoyance.txt
https://raw.githubusercontent.com/cjx82630/cjxlist/master/cjxlist.txt
https://fanboy.co.nz/r/fanboy-complete.txt
https://adguard.com/en/filter-rules.html?id=11
```

None of those are formatted as hosts files, and are adding a lot of invalid entries into your `gravity.list` which may be causing some issues. I've loaded your `adlists.list` into my own installation, and haven't noticed any issues so far, but you never know... What I'm saying though is that I don't think those lists are going to add anything of value to your experience..

Secondly, `tieba.baidu.com` is not blocked! It is not in any of the default lists, though it does appear as a rule in [this list that you added](https://raw.githubusercontent.com/cjx82630/cjxlist/master/cjxlist.txt) resulting in some invalid host entries in `gravity.list`:

```
pi@raspberrypi:~ $ cat /etc/pihole/gravity.list | grep tieba.baidu.com
192.168.1.253 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.bonus_forum_aside
fe80::49d7:ea2e:28fd:c0b9 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.bonus_forum_aside
192.168.1.253 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.dialogJmodal
fe80::49d7:ea2e:28fd:c0b9 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.dialogJmodal
192.168.1.253 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.dialogJ.umoney-promotion-dialog
fe80::49d7:ea2e:28fd:c0b9 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.dialogJ.umoney-promotion-dialog
192.168.1.253 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com###forum_recommend
fe80::49d7:ea2e:28fd:c0b9 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com###forum_recommend
192.168.1.253 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com###pagelet_frs-aside\/pagelet\/hottopic
fe80::49d7:ea2e:28fd:c0b9 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com###pagelet_frs-aside\/pagelet\/hottopic
192.168.1.253 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com###questionnaire_bg
fe80::49d7:ea2e:28fd:c0b9 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com###questionnaire_bg
192.168.1.253 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com###questionnaire_contianer
fe80::49d7:ea2e:28fd:c0b9 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com###questionnaire_contianer
192.168.1.253 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.thread_recommend
fe80::49d7:ea2e:28fd:c0b9 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.thread_recommend
192.168.1.253 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.topic_list_box
fe80::49d7:ea2e:28fd:c0b9 1111.baidu.com,jump.bdimg.com,tieba.baidu.com,tieba.com,xingqu.baidu.com##.topic_list_box
```

Now, even if I do whitelist `tieba.baidu.com`, it doesn't make a difference, because it does not appear in `gravity.list` to start with, except on those lines above that are, to all intents and purposes, gibberish!

I think you may have an additional problem somewhere along the line, because I can access `tieba.baidu.com` just fine on my network, even with the lists you've added in. Pinging it gives the following results:

```
pi@raspberrypi:~ $ ping tieba.baidu.com
PING post.n.shifen.com (103.235.46.139) 56(84) bytes of data.
64 bytes from 103.235.46.139: icmp_seq=1 ttl=48 time=240 ms
64 bytes from 103.235.46.139: icmp_seq=2 ttl=48 time=240 ms
```

What do you get when you ping that domain?

If I were you, I'd delete `/etc/pihole/adlists.list` and work with the defaults

**Comment by gerard33** [comment]
Whitelisting and blacklisting is working fine now with v2.92 on my Ubuntu Server 14.04 with PHP7.
